spec/acceptance/node_classification_spec.rb

```diff
@@ -36,19 +38,21 @@
         'File',
         "#{TEST_ENVIRONMENT_CODE_DIR}/modules",
         ensure: 'link',
-        target: '/etc/puppetlabs/code/environments/production/modules',
+        target: "#{CODE_DIR_BASE_PATH}/production/modules",
       ),
       declare(
         'node_group',
         TEST_ENVIRONMENT,
         ensure: 'present',
-        parent: 'All Environments',
+        parent: 'All Nodes',
         environment: TEST_ENVIRONMENT,
         override_environment: true,
       ),
     )
     master.apply_manifest(setup_manifest)
-    master.run_shell("puppet task run servicenow_cmdb_integration::add_environment_rule --params '{\"group_names\": [\"#{TEST_ENVIRONMENT}\"]}' --nodes #{master.uri}")
+    master_hostname = CMDBHelpers.service_name(master) || master.uri
```

Comment:
So the only thing this container stuff should affect is setup or any acceptance test helpers. When I'm writing my acceptance tests, I shouldn't care about whether my master's a container or VM.

That means we should still be using `master.uri` (and `master` everywhere else we are doing stuff on the master). Your container stuff should still work _if_ you include a `roles: ['master']` key under the PE master container's `vars` when generating the inventory file.

Reply:
Unfortunately, there is a difference between the host name from the external perspective of the docker transport, which is what you have to put in the inventory.yaml as the uri, and the host name internal to the container. In this case, the uri name in the inventory is `acceptance_puppet_1`, which is the name you have to use to connect to the container from the outside. One you are inside the container however the `hostname` is simply `puppet`. If I can get the `hostname` directive in the `docker-compose.yaml` file to accept `acceptance_puppet_1` without the whole environment falling apart, then I'll do that, but my initial attempts resulted in everything breaking. I'll probably try again at some point as a refinement.

Reply:
That sounds like a `hostname` method in the `Target` class whose implementation is something like
```
def hostname
  if vas['hostname'] exists
    // use that
  else
    uri
  end
end
```